store/smt.go

```diff
@@ -419,6 +428,165 @@ func (s *SMT) validateTarget() lib.ErrorI {
 	return nil
 }
 
+// GetMerkleProof() returns the merkle proof-of-membership for a given key if it exists,
+// and the proof of non-membership otherwise
+func (s *SMT) GetMerkleProof(key []byte) ([]*lib.Node, lib.ErrorI) {
+	// calculate the key and value to traverse
+	s.target = &node{Key: newNodeKey(crypto.Hash(key), s.keyBitLength)}
+	// check to make sure the target is valid
+	if err := s.validateTarget(); err != nil {
+		return nil, err
+	}
+
+	proof := make([]*lib.Node, 0)
+
+	// navigates the tree downward
+	if err := s.traverse(); err != nil {
+		return nil, err
+	}
+
+	// Add current to the list of traversed nodes (actual node in case of proof of membership),
+	// possible location in the case of proof of non-membership
+	s.traversed.Nodes = append(s.traversed.Nodes, s.current.copy())
```

Comment:
I'm worried this will panic if `s.current` is nil

Reply:
Although upon second check, I'm not sure current can be nil because the traversal should return an error.

Is there a case where `s.current` could be nil during a proof of `non-membership`?

Reply:
no, because even in non membership on a default tree (root, min and max) on non membership `current` will be equal to either min or max depending on where it would be added